### PR TITLE
[Mellanox] Implement new platform API for SONiC physical entity mib extension

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
**- Why I did it**

In order to support SONiC physical entity mib extension, a few new platform API are added to sonic-platform-common, this PR is to provide an mellanox platform implementation for those new APIs.

**- How I did it**

Implement all new APIs.

**- How to verify it**

Manual tests on MSN2410.

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 1]
```diff
@@ -495,3 +495,19 @@ def get_status_led(self):
...
         """
         return None if not Chassis._led else Chassis._led.get_status()
+
+    def get_position_in_parent(self):
+        """
+        Retrieves 1-based relative physical position in parent device
+        Returns:
+            integer: The 1-based relative physical position in parent device
+        """
+        return -1
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:505]: "-1" as a special indicator shall be documented in the description of the API definition.

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 2]
```diff
@@ -294,6 +295,10 @@ def __init__(self, sfp_index, sfp_type):
         self.sdk_handle = None
         self.sdk_index = sfp_index
 
+        # initialize SFP thermal list
+        from .thermal import initialize_sfp_thermals
+        initialize_sfp_thermals(platform, self._thermal_list, self.index)
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:300]: do we need this? Is it possible to keep using the thermal value in the current TRANSCEIVER_DOM_INFO table?

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 3]
```diff
@@ -2149,3 +2154,19 @@ def set_power_override(self, power_override, power_set):
             return self._write_i2c_via_mcia(0, 0x50, MCIA_ADDR_POWER_OVERRIDE, power_set_bit|power_override_bit, power_override_mask)
         else:
             return NotImplementedError
+
+    def get_position_in_parent(self):
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:2158]: interface index is enough? do we need this?

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 2 above)
Comment (Junchao-Mellanox) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:300]: Yes, we need this. TRANSCEIVER_DOM_SENSOR table does not contains all the information that TEMPERATURE_INFO table needs. For example, warning status, critical_high_threshold and critical_low_threshold.

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py (hunk 1 above)
Comment (Junchao-Mellanox) [reply platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:506]: fixed

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 3 above)
Comment (Junchao-Mellanox) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: Fixed

**Review by keboliu — APPROVED** [review]

**Comment by Junchao-Mellanox** [comment]
https://github.com/Azure/sonic-buildimage/pull/5645

**Pull request closed without merging**